src/Components/TheoryPage.js

```diff
@@ -0,0 +1,989 @@
+        },
+      ],
+    },
+    {
+      id: 13,
+      title: 'Z',
+      data: [
+        {
+          id: 0,
+          text: 'Zone',
+          definition:
+            'There are two types of defensive tactics in Ultimate, zone and man defense. Zone is when you guard a specific portion of the field while man defense focuses on guarding a specific player. The most effective form of zone defense in Ultimate is known as the Cup.',
+        },
+      ],
+    },
+  ];
+
+  const extractKey = ({ id }) => id;
+
+  const showModal = selectedItem => {
+    setModalVisible(true);
+  };
+
+  const _onPressItem = item => {
+    setSelectedItem(item);
+    showModal(selectedItem);
+  };
+
+  const _renderItem = ({ item }) => {
+    return (
+      <View>
+        <TouchableHighlight
+          onPress={() => {
+            _onPressItem({ item });
+          }}
+        >
+          <Text style={styles.row}>{item.text}</Text>
+        </TouchableHighlight>
+      </View>
+    );
+  };
+
+  const displayDictionary = () => {
+    return (
+      <View style={styles.container}>
+        <SectionList
+          style={styles.container}
+          sections={sections}
+          renderItem={item => _renderItem(item)}
+          renderSectionHeader={renderSectionHeader}
+          keyExtractor={extractKey}
+        />
+        <View>
+          <Modal
+            animationType="slide"
+            visible={modalVisible}
+            transparent
+            onRequestClose={() => {
+              setModalVisible(!modalVisible);
+            }}
+          >
+            <View style={styles.centeredView}>
+              <View style={styles.modalView}>
+                <Text style={styles.modalTitle}>{selectedItem.item.text}</Text>
+                <Text style={styles.modalText}>{selectedItem.item.definition}</Text>
+                <TouchableHighlight
+                  style={styles.returnButtonDictionnary}
+                  onPress={() => {
+                    setModalVisible(!modalVisible);
+                  }}
+                >
+                  <Text style={styles.textStyle}>Return</Text>
+                </TouchableHighlight>
+              </View>
+            </View>
+          </Modal>
+        </View>
+      </View>
+    );
+  };
+
+  const renderSectionHeader = ({ section }) => {
+    return <Text style={styles.header}>{section.title}</Text>;
+  };
+
+  return (
+    <View style={styles.container}>
+      <View>
+        <View style={styles.containerDictionary}>
+          <View>
+            <View style={styles.container} width={screenDimension.width}>
+              {displayDictionary()}
+            </View>
+          </View>
+        </View>
+      </View>
+    </View>
+  );
+};
+
+const Essential = () => {
```

Comment:
You should add a `theory` subfolder and put each of these 3 components in a separate component there